kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -0,0 +1,258 @@
+        {
+            add("byte");
+            add("short");
+            add("integer");
+            add("long");
+            add("float");
+            add("double");
+            add("date");
+            add("timestamp");
+            add("string");
+        }
+    };
+
+    /**
+     * Returns true if the given data type is eligible for MIN/MAX data skipping.
+     */
+    private static boolean isSkippingEligibleDataType(DataType dataType) {
+        return skippingEligibleTypeNames.contains(dataType.toString()) ||
+            dataType instanceof DecimalType;
+    }
+
+    /**
+     * Given a data schema returns the expected schema for a min or max statistics column. This
+     * means 1) replace logical names with physical names 2) set nullable=true 3) only keep stats
+     * eligible fields (i.e. don't include fields with isSkippingEligibleDataType=false)
+     */
+    private static StructType getMinMaxStatsSchema(StructType dataSchema) {
+        List<StructField> fields = new ArrayList<>();
+        for (StructField field : dataSchema.fields()) {
+            if (isSkippingEligibleDataType(field.getDataType())) {
+                fields.add(new StructField(
+                    getPhysicalName(field), field.getDataType(), true));
+            } else if (field.getDataType() instanceof StructType) {
+                fields.add(
+                    new StructField(getPhysicalName(field),
+                        getMinMaxStatsSchema((StructType) field.getDataType()), true)
+                );
+            }
+        }
+        return new StructType(fields);
+    }
+
+    /**
+     * Given a data schema returns the expected schema for a null_count statistics column. This
+     * means 1) replace logical names with physical names 2) set nullable=true 3) use LongType
+     * for all fields
+     */
+    private static StructType getNullCountSchema(StructType dataSchema) {
+        List<StructField> fields = new ArrayList<>();
+        for (StructField field : dataSchema.fields()) {
+            if (field.getDataType() instanceof StructType) {
+                fields.add(
+                    new StructField(getPhysicalName(field),
+                        getNullCountSchema((StructType) field.getDataType()), true)
+                );
+            } else {
+                fields.add(new StructField(
+                    getPhysicalName(field), LongType.LONG, true));
+            }
+        }
+        return new StructType(fields);
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Public static fields and methods
+    //////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Returns true if the given literal is skipping-eligible. Delta tracks min/max stats for a
+     * limited set of data types and only literals of those types are skipping eligible.
+     */
+    public static boolean isSkippingEligibleLiteral(Literal literal) {
```

Comment:
why this single line function being called from only one location?

Reply:
It will be used more once we add additional expressions to the skipping